doc/contrail_deployment/CAD_for_windows_design.md

```diff
@@ -0,0 +1,50 @@
+# Contrail-ansible-deployer for Windows design document
+
+## Supported orchestrators:
+  * `Openstack` - keystone is needed by Windows docker driver
+
+## Playbooks:
+  * `provision_instances.yml` - not supported at this moment
+  * `configure_instances.yml`
+  * `install_openstack.yml`
+  * `install_contrail.yml`
+
+## OS-specific roles:
+  * There are particular roles which needs to be done differently for Windows
+  * Thus, these roles are divided into `*_Linux` and `*_Win32NT` for usage in c-a-d for Linux and c-a-d for Windows respectively
+
+## Configure_instances:
+  * Some differences between Windows and non-Windows c-a-d workflow exist for this playbook:
+    * Windows compute nodes need Windows-specific dependencies and particular features turned on/off. `Install_software_Win32.yml` is responsible for that
+    * In the future, vRouter kernel module will be digitally signed, but for now, it uses self-signed certificates.
+    To do that, Windows Test-Signing Mode must be enabled and it can only be done after reboot
+
+## Install_openstack:
+  * There aren't any changes to this playbook
+  * It is needed, because Windows docker driver uses keystone for authorization
+  * Whole openstack is installed for now, but in the future there should be need to specify only the role in `config/instances.yaml` which is responsible for installing keystone
+
+## Install_contrail:
+  * For debugging, specific dlls need to be present in main Windows system directory (`C:/Windows/System32`)
+  * Because on Windows only contrail compute nodes are supported, there are 2 supported roles:
+    * `vrouter` - installs vRouter kernel module, vRouter agent and utils (`create_vrouter_Win32NT.yml`)
+      * Pulls artifacts from `contrail-windows-vrouter` image. Artifacts' directory structure:
+        * `C:/Artifacts/`
+          * `agent/`
+            * `contrail-vrouter-agent.msi`
+          * `vrouter/`
+            * `utils.msi`
+            * `vRouter.msi`
+            * `vRouter.cer`
+          * `vtest/*` - vtest utility (needed in Windows CI)
+    * `win_docker_driver` - installs Windows docker driver (`create_win_docker_driver.yml`)
+      * Pulls artifact from `contrail-windows-docker-driver` image. Artifact's directory structure:
+        * `C:/Artifacts/`
+          * `docker_driver/docker-driver.msi`
+  * On Windows contrail-ansible-deployer starts components as Windows services,
+    in contrast to Linux where they reside in separate containers, because:
+    * No possibility to run Linux containers on Windows Server 2016
+    * It isn't tested yet, but there are possible issues:

```

Comment:
> It isn't tested yet, but there are possible issues:

"But" kind of negates the previous part of the sentence, but it doesn't in this case, which is kind of weird to read. Makes you think this is an error.

Maybe we should restructure this paragraph anyways, to something like:
```
... , because:
* No possibility to run Linux containers on Windows Server 2016
* On other Windows versions, containerization of Windows Compute components wasn't tested and implemented yet
* There are plans to do it in future release

Note: potential challenges:
* Security
* Communication
```